File: Algorithms/708. Insert into a Sorted Circular Linked List/php/solution.php
<?php

/**
 * Definition for a Node.
 * class Node {
 *     public $val = null;
 *     public $next = null;
 *     function __construct($val = 0) {
 *         $this->val = $val;
 *         $this->next = null;
 *     }
 * }
 */

class Solution
{
    /**
     * @param Node $root
     * @param Integer $insertVal
     * @return Node
     */
    function insert($head, $insertVal)
    {
        if (is_null($head)) {
            $node = new Node($insertVal);
            $node->next = $node;
            return $node;
        }

        $prev = $head;
        $curr = $head->next;
        while ($curr != $head) {
            if ($prev->val <= $insertVal && $insertVal <= $curr->val) {
                break;
            }
            if ($prev->val > $curr->val && ($insertVal >= $prev->val || $insertVal <= $curr->val)) {
                break;
            }
            $prev = $curr;
            $curr = $curr->next;
        }

        $node = new Node($insertVal);
        $node->next = $curr;
        $prev->next = $node;
        return $head;
    }
}

File: Algorithms/708. Insert into a Sorted Circular Linked List/golang/solution.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Next *Node
 * }
 */

func insert(aNode *Node, x int) *Node {
	if aNode == nil {
		newNode := Node{x, nil}
		newNode.Next = &newNode
		return &newNode
	}

	var prev, curr *Node = aNode, aNode.Next
	for curr != aNode {
		if prev.Val <= x && x <= curr.Val {
			break
		}
		if prev.Val > curr.Val {
			if x >= prev.Val || x <= curr.Val {
				break
			}
		}
		prev, curr = curr, curr.Next
	}

	prev.Next = &Node{x, curr}
	return aNode
}